Mathlib/LinearAlgebra/UnitaryGroup.lean

```diff
@@ -226,4 +253,26 @@ theorem mem_orthogonalGroup_iff' {A : Matrix n n β} :
 
 end OrthogonalGroup
 
+section specialOrthogonalGroup
+
+variable (n) (β : Type v) [CommRing β] (A : Matrix n n β)
+
+attribute [local instance] starRingOfComm
+
+/-- `Matrix.specialOrthogonalGroup n` is the group of orthogonal `n` by `n` where the determinant
+is one. (This definition is only correct if 2 is invertible.)-/
+abbrev specialOrthogonalGroup := specialUnitaryGroup n β
+
+theorem mem_specialOrthogonalGroup_iff  {A : Matrix n n β}:
+    A ∈ specialOrthogonalGroup n β ↔ A * star A = 1 ∧ A.det = 1 := by
+  rw [specialOrthogonalGroup]
+  apply mem_specialUnitaryGroup_iff
```

Comment:
If you made `n`, `α`, and `β` implicit (using curly braces after the abbrev), then this could be `exact` rather than `apply`.

Reply:
How would I do this?
`abbrev specialUnitaryGroup {n : Type u} {α : Type v} [Fintype n] [DecidableEq n] [CommRing α] [StarRing α]
  := unitaryGroup n α ⊓ MonoidHom.mker detMonoidHom` This gives me errors in the next lemma

Reply:
Sorry, I meant that you should make `n`, `α`, and `β` implicit a line or two after the `abbrev`. That way they are explicit for the definition but implicit theorems where they can be inferred from `A`.

Reply:
For example:
```
section specialUnitaryGroup

variable (n) (α)

/--`Matrix.specialUnitaryGroup` is the group of unitary `n` by `n` matrices where the determinant
is 1. (This definition is only correct if 2 is invertible.)-/
abbrev specialUnitaryGroup := unitaryGroup n α ⊓ MonoidHom.mker detMonoidHom

variable {n} {α}

theorem mem_specialUnitaryGroup_iff :
    A ∈ specialUnitaryGroup n α ↔ A * star A = 1 ∧ A.det = 1:= by
  rw [← mem_unitaryGroup_iff]
  rfl

end specialUnitaryGroup
```

Reply:
Ok, thank you very much for this help